constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,256 @@ func TestRegoRewriterErrorCases(t *testing.T) {
 		}
 	}
 }
+
+func TestRegoRewriterRewriteDataRef(t *testing.T) {
+	tests := []struct {
+		name    string
+		prefix  string
+		imports string
+		extern  []string
+		want    string
+	}{
+		{
+			name:    "no rewrite when data ref is an external reference",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext", "data.inventory"},
+			want:    "data.ext",
+		},
+		{
+			name:    "rewrite for ref is sub of external references",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext.util"},
+			want:    "data.foo.bar.ext",
+		},
+		{
+			name:    "no rewrite for non data ref",
+			prefix:  "foo.bar",
+			imports: "invalid_data.name",
+			want:    "invalid_data.name",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.some_ref",
+			want:    "data.data.lib.some_ref",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			externs := []string{"data.inventory"}
+			if tt.extern != nil {
+				externs = tt.extern
+			}
+			r, err := New(NewPackagePrefixer(tt.prefix), []string{"data.lib"}, externs)
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
```

Comment:
Nit: It's always fine to just put `t.Fatal(err)`. If you find adding information to the error messages, make them explicit.  "Failed to create" doesn't say much. Instead, use a form like one of the below:

```
t.Fatalf("got New() error = %v, want nil", err)
```
or
```
t.Fatalf("failed to create RegoRewriter: %v", err)
```
The first example form is nice since it explicitly calls out the function which unexpectedly returned an error. The second is nice because it gives the context of what part of the test setup actually failed. It's also fine to combine the two, but it isn't necessary.

Reply:
Updated the failed info for RegoRewriter creation failure.